server/peers.py

```diff
@@ -143,7 +147,7 @@ def on_features(self, features, error):
                 self.peer.update_features(features)
         # For legacy peers not implementing features, check their height
         # as a proxy to determining they're on our network
-        if not verified:
+        if not verified and not self.failed:
             self.send_request(self.on_headers, 'blockchain.headers.subscribe')
```

Comment:
OK; this line is good; I'll put that in my next update.   I'm not sure what the rest is trying to fix or what you saw; can you elaborate?